pkg/cmd/clients.go

```diff
@@ -130,68 +134,165 @@ func (cc *ClientCmd) CreateClientCmd() *cobra.Command {
+		The available client types are android, cordova, iOS and xamarin.
 
-When used standalone, a namespace must be specified by providing the --namespace flag.`,
+		When used standalone, a namespace must be specified by providing the --namespace flag.`,
 		Example: `  mobile create client <name> <clientType> <appIdentifier> --namespace=myproject 
-  kubectl plugin mobile create client <name> <clientType> <appIdentifier>
-  oc plugin mobile create client <name> <clientType> <appIdentifier>`,
+  					kubectl plugin mobile create client <name> <clientType> <appIdentifier>
+					oc plugin mobile create client <name> <clientType> <appIdentifier>`,
 
 		RunE: func(cmd *cobra.Command, args []string) error {
 			if len(args) != 3 {
 				return cmd.Usage()
 			}
+
 			name := args[0]
 			clientType := args[1]
 			appIdentifier := args[2]
-			appKey := uuid.NewV4().String()
-
 			namespace, err := currentNamespace(cmd.Flags())
 			if err != nil {
 				return errors.Wrap(err, "failed to get namespace")
 			}
-			app := &v1alpha1.MobileClient{
-				TypeMeta: metav1.TypeMeta{
-					Kind:       "MobileClient",
-					APIVersion: "mobile.k8s.io/v1alpha1",
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Labels:      map[string]string{},
-					Annotations: map[string]string{},
-				},
-				Spec: v1alpha1.MobileClientSpec{Name: name, ApiKey: appKey, ClientType: clientType, AppIdentifier: appIdentifier},
+
+			clientId := name + "-" + clientType
+			client, _ := cc.mobileClient.MobileV1alpha1().MobileClients(namespace).Get(clientId, metav1.GetOptions{})
+			if client.ObjectMeta.UID != "" {
+				return errors.New("App with this name already exist for this client type")
 			}
-			switch app.Spec.ClientType {
+
+			var apbName string
+			switch clientType {
 			case "android":
-				app.Annotations["icon"] = "fa fa-android"
+				apbName = "android-app"
 				break
 			case "iOS":
-				app.Annotations["icon"] = "fa fa-apple"
+				apbName = "ios"
 				break
 			case "cordova":
-				app.Annotations["icon"] = "font-icon icon-cordova"
+				apbName = "cordova"
 				break
 			case "xamarin":
-				app.Annotations["icon"] = "font-icon icon-xamarin"
+				apbName = "xamarin"
 				break
-			default:
-				return errors.New("Unknown client type")
+
 			}
-			app.Name = name + "-" + strings.ToLower(app.Spec.ClientType)
-			if err := input.ValidateMobileClient(app); err != nil {
-				return errors.Wrap(err, "failed validation while creating new mobile client")
+			//Get available provision parameters from the cluster service plan
+			clusterServiceClass, err := findServiceClassByName(cc.scClient, apbName)
+			if err != nil {
+				return errors.WithStack(err)
 			}
-			createdClient, err := cc.mobileClient.MobileV1alpha1().MobileClients(namespace).Create(app)
+			clusterServicePlan, err := findServicePlanByNameAndClass(cc.scClient, "default", clusterServiceClass.Name)
 			if err != nil {
-				return errors.Wrap(err, "failed to create mobile client")
+				return errors.WithStack(err)
```

Comment:
Might be better to wrap this and the above error like with the previous. I feel we are exposing more than needed to the caller here, wdyt?